Python/Product/PythonTools/PythonTools/Refactoring/LocationPreviewItem.cs

```diff
@@ -30,66 +32,127 @@ namespace Microsoft.PythonTools.Refactoring {
-        }
 
-        public int Line {
-            get {
-                return _lineNo;
+            if (start < 0 || length <= 0) {
+                Debug.Fail("Expected valid span");
+                return;
             }
+
+            _text = text.TrimStart(_whitespace);
+            Line = locationInfo.Line;
+            Column = start + 1;
+            _span = new Span(start - (text.Length - _text.Length), length);
         }
 
-        public int Column {
-            get {
-                return _columnNo;
+        private static bool GetSpan(string text, string origName, AnalysisLocation loc, out int start, out int length) {
+            if (string.IsNullOrEmpty(text)) {
+                throw new ArgumentNullException(nameof(text));
+            }
+            if (string.IsNullOrEmpty(text)) {
+                throw new ArgumentNullException(nameof(origName));
             }
-        }
 
-        public int Length {
-            get {
-                return _span.Length;
+            start = loc.Column - 1;
+            length = origName.Length;
+            if (start < 0 || length <= 0) {
+                Debug.Fail("Invalid span for '{0}': [{1}..{2})".FormatInvariant(origName, start, start + length));
+                return false;
+            }
+
+            var cmp = CultureInfo.InvariantCulture.CompareInfo;
+            try {
+                if (cmp.Compare(text, start, length, origName, 0, origName.Length) == 0) {
+                    // Name matches, so return the span
+                    return true;
+                }
+            } catch (ArgumentOutOfRangeException ex) {
+                Debug.Fail(ex.ToUnhandledExceptionMessage(typeof(LocationPreviewItem)));
             }
+
+            start = -1;
+            length = -1;
+            return false;
         }
 
-        public VariableType Type {
-            get {
-                return _type;
+        private static bool GetSpanWithPrefix(string text, string origName, AnalysisLocation loc, string prefix, string newName, out int start, out int length) {
+            if (string.IsNullOrEmpty(prefix)) {
+                throw new ArgumentNullException(nameof(prefix));
+            }
+            if (string.IsNullOrEmpty(newName)) {
+                throw new ArgumentNullException(nameof(prefix));
             }
+
+            if (!GetSpan(text, prefix + origName, loc, out start, out length)) {
+                start = -1;
+                length = -1;
+                return false;
+            }
+
+            if (newName.StartsWith("__") && newName.Length > 2) {
+                // renaming from private name to private name, so just rename the non-prefixed portion
+                start += prefix.Length;
+                length -= prefix.Length;
+            }
+
+            return true;
         }
 
+        public int Line { get; }
+        public int Column { get; }
+        public VariableType Type { get; }
+
+        public int Length => _span.Length;
```

Comment:
nevermind, I just noticed it's a struct.

---
In reply to: [109052129](https://github.com/Microsoft/PTVS/pull/2341#discussion_r109052129) [](ancestors = 109052129)